share/smack/prelude.py

```diff
@@ -0,0 +1,129 @@
+class Function:
+  def __init__(self, attr, attr_arg, name, arg_types, ret_type, body):
+    self.attr = attr
+    self.attr_arg = attr_arg
+    self.name = name
+    self.arg_types = arg_types
+    self.ret_type = ret_type
+    self.vars = map(lambda i : 'i' + str(i), range(len(arg_types)))
+    self.body = body
+
+  def toStr(self):
+    func_str = 'function '
+    if self.attr is not None:
+      func_str += '{:' + self.attr
+      if self.attr_arg is not None:
+        func_str += ' \"' + self.attr_arg + '\"'
+      func_str += '} '
+    func_str += self.name + '('
+    for i in range(len(self.arg_types)):
+      func_str += 'i' + str(i + 1) + ': ' + self.arg_types[i]
+      if i < len(self.arg_types) - 1:
+        func_str += ', '
+    func_str += ') returns (' + self.ret_type + ')'
+    if self.body is not None:
+      func_str += ' {'+ self.body + '}'
+    else:
+      func_str += ';'
+    func_str += '\n'
+    return func_str
+
+mk_int = lambda bitwidth: 'i' + str(bitwidth)
+mk_bv = lambda bitwidth: 'bv' + str(bitwidth)
+
+UNINTERPRETED_UNARY_OP = lambda bitwidth, name: Function(None, None, '.'.join([name, mk_int(bitwidth)]), [mk_int(bitwidth)], mk_int(bitwidth), None)
```

Comment:
Could you format this to look nicer and more readable somehow? Maybe split arguments to Function into multiple lines?